src/lib/featureFlags/unleashProvider.ts

```diff
@@ -0,0 +1,83 @@
+import { startUnleash, Unleash } from "unleash-client"
+import {
+  UNLEASH_API,
+  UNLEASH_APP_NAME,
+  UNLEASH_INSTANCE_ID,
+  UNLEASH_SERVER_KEY,
+} from "../../config"
+import { FeatureFlagConfig, FeatureFlagProvider } from "./featureFlagProvider"
+
+export const UnleashProvider = Symbol("UnleashProvider")
+
+export class UnleashProviderConfig implements FeatureFlagConfig {
+  constructor(
+    private _url: string,
+    private _appName: string,
+    private _instanceId: string,
+    private _serverKey: string
+  ) {}
+
+  [Symbol.iterator] = function* (): Iterator<string> {
+    yield this.url
+    yield this.appName
+    yield this.instanceId
+    yield this.serverKey
```

Comment:
What is the reasoning behind using such advanced JS features as generators here?

Reply:
Have removed the custom config and generator.